File: app/src/main/java/com/davismiyashiro/expenses/view/opentab/ReceiptInterfaces.java
package com.davismiyashiro.expenses.view.opentab;

import android.support.v4.util.ArrayMap;

import com.davismiyashiro.expenses.datatypes.ReceiptItem;
import com.davismiyashiro.expenses.datatypes.Tab;

import java.util.List;

/**
 * @Author Davis Miyashiro
 */

public interface ReceiptInterfaces {
    interface ReceiptView{

        void showReceiptItems (ArrayMap<String, List<ReceiptItem>> items);

        void refreshReceiptItemsAdapter ();
    }

    interface UserActionsListener {

        void setReceiptView (ReceiptView view);

        void loadReceipItems (Tab tab);
    }
}

File: app/src/main/java/com/davismiyashiro/expenses/view/opentab/ReceiptInterfaces.kt
/*
 * MIT License
 *
 * Copyright (c) 2019 Davis Miyashiro
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */
package com.davismiyashiro.expenses.view.opentab

import android.support.v4.util.ArrayMap

import com.davismiyashiro.expenses.datatypes.ReceiptItem
import com.davismiyashiro.expenses.datatypes.Tab

/**
 * @Author Davis Miyashiro
 */

interface ReceiptInterfaces {
    interface ReceiptView {

        fun showReceiptItems(items: MutableMap<String, MutableList<ReceiptItem>>)

        fun refreshReceiptItemsAdapter()
    }

    interface UserActionsListener {

        fun setReceiptView(view: ReceiptView)

        fun loadReceipItems(tab: Tab)
    }
}
